src/mlpack/core/util/param_checks.hpp

```diff
@@ -206,6 +206,31 @@ void ReportIgnoredParam(
 void ReportIgnoredParam(const std::string& paramName,
                         const std::string& reason);
 
+/**
+ * If the input has correct shapes(kind of santity checks). 
+ *
+ * @code
+ * Number of data points in both X and y of linear regression should be same. Also the features in test and training matrix should be same.
+ * @endcode
+ *
+ * Obviously for all the data points we must have both X and y for all the points in linear regression. 
+ *
+ * @param method Name of the method for which santity task is to be performed.
+ * @param x_rows the number of rows in training matrix X
+ * @param x_cols the number of columns in training matrix X
+ * @param y_rows the number of rows in training label matrix y
+ * @param y_cols the number of rows in training label matrix y
+ * @param xt_rows the number of rows in test matrix
+ * @param xt_cols the number of rows in test matrix
+ * 
+ * 
+ */
+
+void CheckInputShape(const size_t x_rows,
```

Comment:
Take a look at #2370, which has some utilities like this.  A better interface might be a function internal to `IO` that takes the names of two matrix parameters and checks that, e.g., the rows are the same or the columns are the same or similar.

Reply:
Also note that @onikolskyy is working on similar support in #2847, so it might be worthwhile to coordinate and make just one utility function (or set of utility functions) that can be used in both cases.